Automattic/wp-calypso
# Site Transfers: restyle Domains section on transfer screen

Closes https://github.com/Automattic/dotcom-forge/issues/2625

## Proposed Changes

This PR restyles **Domains** section on the transfer site screen to be more consistent with Figma design file: 1U910ArebC7gbLF1RtxUxE-fi-1_2.

**Before**:

<img width="752" alt="Screenshot 2023-06-19 at 4 04 45 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/60040af4-0643-4be5-a4ca-9fd5d86ae53f">

**After**: 

<img width="788" alt="Screenshot 2023-06-19 at 4 03 43 PM" src="https://github.com/Automattic/wp-calypso/assets/25575134/79568f2d-9e0a-4d46-a412-349e9b8c2c6c">


## Testing Instructions

* Make sure you have a site with at least one custom domain
* Navigate to **Settings > General > Transfer your site**: https://wpcalypso.wordpress.com/settings/start-site-transfer/{yoursite}
* Select an administrator to transfer your site to
* Proceed to the transfer screen
* Observe the design of the "Domains" section

**Note**: In my PR, the domain icons are not fully aligned with the text but this was the best I could do with them. If you have any suggestions on aligning them better, let me know. 

## Pre-merge Checklist
